src/types/pkl.ts

```diff
@@ -1,3 +1,47 @@
+export type Any = null | AnyObject | Map<Any, Any> | string | number | bigint | boolean
+export type AnyObject = BaseObject | Map<Any, Any> | Any[] | Set<Any> | Duration | DataSize | Pair<any, any> | IntSeq | Regex | {}
+
+// BaseObject is the TS representation of `pkl.base#Object`.
+// It is a container for properties (as object properties), entries (in an interior Map), and elements (as Array elements).
+export class BaseObject extends Array<any> {
```

Comment:
We might want to define a TS interface for BaseObject, and then have a class satisfy the interface, so that BaseObject can be used as a type.